### PR TITLE
Add Google OAuth

### PR DESCRIPTION
**Pull request by plin-stytch**
Added in Discovery login with Google OAuth.

New functionality:
https://github.com/user-attachments/assets/7a108778-f623-4732-af63-1df96a1a2d75

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -13,30 +13,32 @@ const app = express();
 // Use EJS for HTML templating.
 app.set('view engine', 'ejs');
 
-app.use(express.static(__dirname + "/static"));
+app.use(express.static(__dirname + '/static'));
```
Comment (plin-stytch) [inline server.js:16]: My linter went a little craz. If you'd like to review the actual code changes, I'd suggest hiding whitespace changes at the top here:
Screenshot 2024-10-21 at 11 28 22 AM

File: server.js [hunk 2]
```diff
@@ -164,60 +164,63 @@ app.post('/send-magic-link', async (req, res) => {
...
-            return;
-        }
-
-        req.session[StytchSessionKey] = resp.session_token;
-        res.redirect('/');
-        return;
+    req.session[StytchIstKey] = resp.intermediate_session_token;
+    responseDiscoveredOrganizations = resp.discovered_organizations;
+    authenticatedEmail = resp.email_address;
+  } else if (tokenType === 'discovery_oauth') {
```
Comment (plin-stytch) [inline server.js:192]: This case is the main one that was added

File: server.js [hunk 3]
```diff
@@ -164,60 +164,63 @@ app.post('/send-magic-link', async (req, res) => {
...
+      discovery_magic_links_token: token,
+    });
+    if (resp.status_code !== 200) {
+      console.error('Authentication error');
+      res.status(500).send();
+      return;
     }
 
-    // Handle Organization authentication.
-    if (tokenType === 'multi_tenant_magic_links') {
```
Comment (plin-stytch) [inline server.js:204]: This was deleted since we are doing discovery login now

**Review by mary-stytch — APPROVED** [review]
LGTM

**Pull request merged**